actors/builtin/power/power_test.go

```diff
@@ -145,6 +148,115 @@ func TestPowerAndPledgeAccounting(t *testing.T) {
+
+		actor.updateClaimedPower(rt, miner4, powerUnit, mul(powerUnit, 2))
+		actor.updateClaimedPower(rt, miner5, powerUnit, mul(powerUnit, 2))		
+
+		// Below threshold small miner power is counted
+		expectedTotalBelow := big.Sum(mul(smallPowerUnit, 3), mul(powerUnit, 2))
+		actor.expectTotalPower(rt, expectedTotalBelow, mul(expectedTotalBelow, 2))
+
+		// Above threshold (power.ConsensusMinerMinMiners = 3) small miner power is ignored
+		actor.updateClaimedPower(rt, miner3, powerUnit, mul(powerUnit, 2))
+		expectedTotalAbove := big.Sum(smallPowerUnit, mul(powerUnit, 3))
+		actor.expectTotalPower(rt, expectedTotalAbove, mul(expectedTotalAbove, 2))
+
+		st := getState(rt)
+		assert.Equal(t, int64(3), st.NumMinersMeetingMinPower)
+
+		// Less than 3 miners below threshold again small miner power is counted again
+		negPowerUnit := powerUnit.Neg()
+		actor.updateClaimedPower(rt, miner3, negPowerUnit, mul(negPowerUnit, 2))
+		actor.expectTotalPower(rt, expectedTotalBelow, mul(expectedTotalBelow, 2))
+	})
+
+	t.Run("all miner power dissapears when miner dips below min power threshold", func(t *testing.T) {
+		// Setup four miners above threshold
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		actor.createMinerBasic(rt, owner, owner, miner1)
+		actor.createMinerBasic(rt, owner, owner, miner2)
+		actor.createMinerBasic(rt, owner, owner, miner3)		
+		actor.createMinerBasic(rt, owner, owner, miner4)				
+
+		actor.updateClaimedPower(rt, miner1, powerUnit, powerUnit)
+		actor.updateClaimedPower(rt, miner2, powerUnit, powerUnit)
+		actor.updateClaimedPower(rt, miner3, powerUnit, powerUnit)
+		actor.updateClaimedPower(rt, miner4, powerUnit, powerUnit)
+
+		expectedTotal := mul(powerUnit, 4)
+		actor.expectTotalPower(rt, expectedTotal, expectedTotal)
+
+		// miner4 dips just below threshold
+		actor.updateClaimedPower(rt, miner4, smallPowerUnit.Neg(), smallPowerUnit.Neg())
+
+		expectedTotal = mul(powerUnit, 3)
+		actor.expectTotalPower(rt, expectedTotal, expectedTotal)
+	})
+
+	t.Run("threshold only depends on qa power, not raw byte", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		actor.createMinerBasic(rt, owner, owner, miner1)
+		actor.createMinerBasic(rt, owner, owner, miner2)
+		actor.createMinerBasic(rt, owner, owner, miner3)		
+
+		actor.updateClaimedPower(rt, miner1, powerUnit, big.Zero())
+		actor.updateClaimedPower(rt, miner2, powerUnit, big.Zero())
+		actor.updateClaimedPower(rt, miner3, powerUnit, big.Zero())
+		st := getState(rt)
+		assert.Equal(t, int64(0), st.NumMinersMeetingMinPower)
+
+		actor.updateClaimedPower(rt, miner1, big.Zero(), powerUnit)
+		actor.updateClaimedPower(rt, miner2, big.Zero(), powerUnit)
+		actor.updateClaimedPower(rt, miner3, big.Zero(), powerUnit)
+		st = getState(rt)
+		assert.Equal(t, int64(3), st.NumMinersMeetingMinPower)
+	})
+
+	t.Run("slashing miner below minimum does not impact power", func(t *testing.T) {
```

Comment:
This is a little confusing. It's "Slashing a miner who is below minimum", not  "Slashing a miner to go below minimum".